### PR TITLE
[chat][docs] Add docs overview gallery and playground

### PR DESCRIPTION
**Pull request by hasdfa**
This PR builds out the **Chat docs overview gallery** and **playground** experience, hardens the Material message surfaces in `@mui/x-chat`, and follows up with a **docs-accuracy pass** that ships the runtime APIs the docs describe and adds a CI guard so docs code stays in sync with the package.

## Docs

- **Overview gallery** — new `docs/src/modules/chat-gallery/*` component showcase (message, composer, conversation, layout, and state thumbnails) plus a `ChatComponentsSection` and an `all-components` page that links every chat surface.
- **Interactive playground** — new `docs/src/modules/components/chat-playground/*` infrastructure (`PlaygroundCard`, controls, shared fixtures/providers, `parseSx`) wiring **Props + Classes** tabs onto live demos.
- **Overview main demo** — multiple presets (basic, agent, copilot, messenger, widget, captions) with generated code sources and a view toggle.
- **New / updated pages** — `basics/conversation`, `display/date-divider`, `display/unread-marker`, `customization/{overview,structure,look-and-feel}`, `backend/built-in-adapters/{ai-sdk,echo}`, plus refreshed quickstart, basics, behavior, display, and multi-conversation content.
- **API docs** regenerated for the new/changed components and translations.

## `@mui/x-chat` package

- **`ChatMessageError`** — new public error-surface component with its own classes (`chatMessageErrorClasses`, `getChatMessageErrorUtilityClass`).
- **Context-based slot distribution** — new `ChatSlotsContext` and a flat slot vocabulary across the Material message/composer/conversation surfaces.
- **`ChatBox` layout modes** — new `ChatBoxLayoutMode` / `ChatBoxLayoutModeBreakpoints` types and layout slots.
- **Hardened content rendering** — replaced the hand-rolled markdown parser with [`markdown-to-jsx`](https://www.npmjs.com/package/markdown-to-jsx) (`^9.8.1`), with `streamingMarkdownRepair` to gracefully close partial markdown during streaming.
- **New internals (tested)** — `mergeSlotProps`, `useCopyToClipboard`, `streamingMarkdownRepair`, and a `DefaultMessageItem` for `ChatMessageList`.
- **Polish** — unread marker rendered as a centered chip with a divider; `theme.alpha` used for the error card background (CSS-vars safe).

## New runtime APIs

A docs review found several pages documenting APIs (or recipes) that didn't actually exist. Rather than weaken the docs, these were implemented in `@mui/x-chat-headless` and surfaced through `@mui/x-chat`:

- **Regenerate** — `chat.regenerate(messageId)` runtime action (`useChat().regenerate`) plus a new subscription-free **`useChatActions()`** hook, an optional `adapter.regenerate` transport hook, a `REGENERATE_ERROR` code, and `regenerate` bridges for the echo and Vercel AI SDK adapters. Message actions gain **`extraActions`** with function-valued `slotProps.messageActions`.
- **Typing signal** — opt-in **`features.typingSignal`** (off by default) that edge-triggers `adapter.setTyping()` on the composer going empty↔non-empty, via new `ChatFeatures` plumbing through `ChatProvider` / `ChatRoot` / `ChatBox`.
- **`onReachBottom`** on the message list (strict reach semantics mirroring `onReachTop`), plus JSDoc for the existing `onReachTop`.
- **`isLoadingHistory`** — exposed across the store, a `selectIsLoadingHistory` selector, `useChat` / `useChatStatus`, and `ChatPublicState`; true during both initial history fetch and `loadMoreHistory`.
- **`shouldShowDivider`** predicate on the date divider for custom grouping (default by-day behavior unchanged), with by-week / every-N-messages recipes in the docs.
- **`ConversationList.Item`** added to the headless namespace (was documented but only `Root` was exported).

`ChatMessageSkeleton` now respects `prefers-reduced-motion`. All additions are covered by new unit + browser tests (882 jsdom / 908 browser passing).

## Docs correctness

- **New Accessibility page** (nav-reachable) documenting roving focus, Enter drill-in, landmarks, the streaming announcer, and reduced motion, plus an "Accessible by default" overview section.
- **Fixed broken API tables** — union types were mis-escaped (`\ | ` instead of `\|`), splitting cells and adding phantom columns across 16 pages; repaired mechanically.
- **Correctness pass** across the page set — real slot names, actual defaults (`read` status, author grouping, `renderStreamingMarkdown`), `data-${string}` part types, `toolCallId` vs `approvalId`, and completed over-claimed sets (the all-components gallery, the streaming chunk-type table, and the selector/hook reference pages now match the real export surface).
- **Live demos** added to previously code-only feature pages (markdown rendering, files & images, tool calling, tool approval, reasoning, step tracking, read-receipts scroll-to-bottom, layout two-pane, controlled state, adapters).
- **Removed six orphan customization draft pages**; customization docs consolidated on overview / look-and-feel / structure.

## Tooling

- **Docs-correctness guard** — a new vitest suite (`packages/x-chat/src/tests/docsCorrectnessGuard/`) that fails CI if any docs code fence or demo `.tsx` imports a symbol that isn't exported from the stated entry point, or if frontmatter `components:` / `hooks:` reference a missing export. Export surfaces are enumerated via the TypeScript checker; it ships green and prevents this whole class of doc drift from regressing.

## New dependency

- `markdown-to-jsx@^9.8.1` (added to `@mui/x-chat`).

---

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/x-chat-headless/src/internals/useChatController/sendMessageActions.ts [hunk 1]
```diff
@@ -39,6 +39,90 @@ export function createSendMessageActions<Cursor = string>(params: {
...
+        return result;
+      }
+
+      return await processStream(store, reconnectStream, {
+        ...processOptions,
+        messageId: result.messageId,
+        // Allow any replayed sequence number to apply on the reconnected
+        // stream (#5). Without this, chunks the server resends from an
+        // earlier point would be silently dropped by the sequence guard.
+        reconnectFromSequence: 0,
```
[the remark this answers is not in the document]
Comment (hasdfa) [reply packages/x-chat-headless/src/internals/useChatController/sendMessageActions.ts]: Fixed lower in the stack in `1dba6392bf`: reconnect now starts from `result.nextSequence` and carries seen event IDs into the reconnect stream. This branch has been rebased on that fix.

File: packages/x-chat-headless/src/internals/useChatController/useChatController.ts [hunk 2]
```diff
@@ -198,6 +218,82 @@ export function useChatController<Cursor = string>({
...
+  // the (often hundreds of) repeats from a single leaky stream.
+  React.useEffect(() => {
+    const win = (globalThis as unknown as { window?: Window }).window;
+    if (!win) {
+      return undefined;
+    }
+
+    const STREAM_PRODUCER_ERROR_PATTERN =
+      /enqueue.*(closed|cancelled)|(closed|cancelled).*enqueue|invalid state.*(closed|cancelled).*stream/i;
+    const DEDUPE_WINDOW_MS = 3000;
```
[the remark this answers is not in the document]
Comment (hasdfa) [reply packages/x-chat-headless/src/internals/useChatController/useChatController.ts]: Fixed lower in the stack in `1dba6392bf`: the broad window-level `error` / `unhandledrejection` interception was removed, so unrelated page errors are no longer classified as chat stream failures. This branch has been rebased on that fix.

**Review by joserodolfofreitas — COMMENTED** [review]
Nice Vadym, thanks for the refinements!

Some notes:
- Can we hide the customization bar in a drawer? I believe we need to show a really basic shape of the chatbox here. The customization buttons feel like noise here, so we better hide it and leave just a small customization button to open the drawer.

Image

In general, I loved the playground widget, also on other examples. It's just that I believe it's important to give the cleanest overview.

- As well, on Chatbox page, the first demo should be pure too (without `features={{ conversationList: true }}` and then the playground we show second.

- Loved the all components page. But I think we should use the brand blue on the assets

**Review by joserodolfofreitas — APPROVED** [review]
Pre-approval in favor of faster iteration over smaller pieces.

**Comment by hasdfa** [comment]
@codex review

File: packages/x-chat/src/ChatBox/ChatBoxContent.tsx [hunk 3]
```diff
@@ -223,284 +309,619 @@ function DefaultConversationHeader({
...
+            </ChatBoxConversationOverlayPanel>
+          </MUIFocusTrap>
+        </ChatBoxConversationOverlay>
+      )}
+
+      {showSplitConversationList && (
+        <ConversationListComponent
+          variant={variant}
+          {...(slotProps.conversationList ?? {})}
+          slotProps={splitConversationListSlotProps}
```
[the remark this answers is not in the document]
Comment (hasdfa) [reply packages/x-chat/src/ChatBox/ChatBoxContent.tsx:981]: Fixed in `f02a39f35a` (#22482). Confirmed — the pane markers were only on the headless `ConversationListRoot`/`ConversationRoot`, but ChatBoxContent renders the Material `ChatConversationList`/`ChatConversation` wrappers (unmarked), so in split mode with the list visible and no active conversation the lone child fell back to the thread pane. Exported `markChatLayoutPane` and marked the Material wrappers (`ChatConversationList` → conversations, `ChatConversation` → thread) and the narrow `ChatBoxConversationOverlay` (conversations). Marking every direct ChatLayout child keeps pane resolution unambiguous. Added a regression test (split + no active conversation → list is in the conversations pane). Rebased on the fix.

File: packages/x-chat/src/ChatMessage/ChatMessageContent.tsx [hunk 4]
```diff
@@ -695,24 +1132,17 @@ const sourceDocumentPartSlots = {
...
-      slots,
-      slotProps,
-      className,
-      classes: classesProp,
-      partProps: userPartProps,
-      ...other
-    } = props;
-    const classes = useChatMessageUtilityClasses(classesProp);
+    const { slots, slotProps, className, partProps: userPartProps, ...other } = props;
+    const classes = useChatMessageUtilityClasses(undefined);
```
[the remark this answers is not in the document]
Comment (hasdfa) [reply packages/x-chat/src/ChatMessage/ChatMessageContent.tsx]: The dropped `classes` override is intentional (Option B): the sub-parts share the single `MuiChatMessage-*` namespace, so per-sub-part `classes` advertised wrong-prefixed classes in the API docs. Kept on the namespace owners (`ChatMessage`, `ChatMessageError`); styling stays available via `slotProps`/`className` + theme `styleOverrides`, and no demo passes `classes` here. The JS `...other` leak is fixed in `1dd458d12a` (a JS/theme-injected `classes` is now destructured out across the sub-parts); this branch is rebased on it.

File: packages/x-chat/src/ChatBox/ChatBoxContent.tsx [hunk 5]
```diff
@@ -223,284 +309,619 @@ function DefaultConversationHeader({
...
+                    <ScrollToBottomComponent {...(slotProps.scrollToBottom ?? {})} />
+                  )}
+                </React.Fragment>
+              }
+              {...(slotProps.messageList ?? {})}
+            />
+            {showCustomEmptyState && CustomEmptyStateComponent && (
+              <ChatBoxCustomEmptyStateOverlay>
+                <ChatBoxCustomEmptyStateInner>
+                  <CustomEmptyStateComponent {...(slotProps.emptyState ?? {})} />
```
[the remark this answers is not in the document]
Comment (hasdfa) [reply packages/x-chat/src/ChatBox/ChatBoxContent.tsx]: Already fixed in `6a537305`: the custom empty-state path now resolves the `SlotComponentProps` callback form (`customEmptyStateProps = typeof slotProps.emptyState === 'function' ? slotProps.emptyState({}) : ...`) before spreading, so owner-state-driven className/sx/handlers are applied. This branch is rebased on it. Covered by a ChatBox regression test (`resolves a function-form emptyState slotProp`).

File: packages/x-chat/src/ChatMessage/ChatMessageAvatar.tsx [hunk 6]
```diff
@@ -48,16 +47,16 @@ const ChatMessageAvatarStyled = styled('div', {
 const ChatMessageAvatar = React.forwardRef<HTMLDivElement, ChatMessageAvatarProps>(
   function ChatMessageAvatar(inProps, ref) {
     const props = useThemeProps({ props: inProps, name: 'MuiChatMessageAvatar' });
-    const { slots, slotProps, className, classes: classesProp, ...other } = props;
-    const classes = useChatMessageUtilityClasses(classesProp);
+    const { slots, slotProps, className, ...other } = props;
+    const classes = useChatMessageUtilityClasses(undefined);
```
[the remark this answers is not in the document]
Comment (hasdfa) [reply packages/x-chat/src/ChatMessage/ChatMessageAvatar.tsx]: Same intentional Option B change as the ChatMessageContent thread: `ChatMessageAvatar` shares the `MuiChatMessage-*` namespace, so its `classes` override was dropped on purpose (kept on `ChatMessage`/`ChatMessageError`). The `...other` leak for JS consumers is fixed in `1dd458d12a` — a JS/theme-injected `classes` is now destructured out across all six sub-parts. Rebased on it.

File: packages/x-chat/src/ChatMessage/ChatMessageGroup.tsx [hunk 7]
```diff
@@ -159,18 +188,17 @@ const ChatMessageGroup = React.forwardRef<HTMLDivElement, ChatMessageGroupProps>
         messageId={messageId}
         {...other}
         slots={{
-          group: slots?.group ?? ChatMessageGroupStyled,
-          authorName: slots?.authorName ?? ChatMessageGroupAuthorNameStyled,
-          groupTimestamp: slots?.groupTimestamp ?? ChatMessageGroupTimestampStyled,
-          ...slots,
+          group: (slots?.messageGroup ?? ChatMessageGroupStyled) as React.ElementType,
```
[the remark this answers is not in the document]
Comment (hasdfa) [reply packages/x-chat/src/ChatMessage/ChatMessageGroup.tsx:225]: By design: the flat slot vocabulary is the intended public API for `ChatMessageGroup` (`messageGroup`/`messageAuthorName`), which the customization docs document. The old headless `group`/`authorName` keys were never part of `ChatMessageGroup`'s published surface (alpha; never shipped with them) and no demo uses them — consumers wanting those slots compose the headless `MessageGroup` directly. Keeping the flat vocab consistent end-to-end.

**Comment by hasdfa** [comment]
@codex review for correctness regressions, missing tests, security issues, performance risks, and risky behavior changes. Ignore style-only nitpicks unless they hide a real bug.

**Comment by hasdfa** [comment]
@codex review this PR as a senior maintainer. Prioritize:
- real bugs
- broken edge cases
- API/contract regressions
- missing tests for changed behavior
- security/auth/data-leak risks
- performance or bundle-size regressions

Do not comment on formatting, naming, or preferences unless they create maintainability risk.

**Comment by hasdfa** [comment]
@codex review for React correctness, accessibility regressions, state bugs, SSR/hydration issues, bundle-size impact, MUI usage mistakes, visual behavior changes, and missing tests. Ignore cosmetic opinions.

**Comment by hasdfa** [comment]
@codex review for security regressions only: injection, secrets, PII leakage, unsafe dependencies, SSRF, XSS, CSRF, insecure redirects, and unsafe file/network handling.

File: docs/src/modules/components/chat-playground/parseSx.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+export function parseSx(raw: string): ParsedSx {
+  const trimmed = raw.trim();
+  if (!trimmed) {
+    return { value: undefined, error: undefined };
+  }
+  try {
+    // Allow either bare object `{ … }` or expression-wrapped `({ … })`.
+    const body = trimmed.startsWith('{') ? `(${trimmed})` : trimmed;
+    // eslint-disable-next-line no-new-func
+    const result = new Function(`"use strict"; return (${body});`)();
```
[the remark this answers is not in the document]
Comment (hasdfa) [reply docs/src/modules/components/chat-playground/parseSx.ts]: Fixed in `cd469d3694`. You're right — `parseSx` evaluated the editor input via `new Function`, so a pasted snippet would run arbitrary JS in the docs origin (silently, on blur/render). Replaced the eval with a small non-executing recursive-descent parser for a safe subset of object-literal syntax (objects, arrays, strings, numbers, booleans, null; quoted or bare keys; trailing commas; comments). Function/arrow values, calls, bare identifiers, and template literals are now rejected with an error instead of being run — no new dependency. Added tests, including that executable payloads are rejected and the probe is never invoked.
